### DISCUSSION TITLE
Fixed Channel / Frequency for Class A LoRaWAN to a Dragino Gateway with ISP4520?

### DISCUSSION THREAD

**Discussion by FieldDataTech** [Lora-net/LoRaMac-node#1020] [Q&A]
Hello,
I'm trying to use the Class A LoRaWAN example to send through a Dragino LG02 gateway to TTN from an ISP4250 eval kit. The Class A code hops frequencies but the Dragino LoRaWAN LG02 Gateway has to be hard configured for a single channel. Is there a way to configure the Class A code to send only on one hard-coded channel?
I'm seeing bits of code referencing a channel mask but I can't find it. Thanks! -Doug

**Comment by FieldDataTech** [comment]
...I might have found my own answer. There's some code in RegionUS915InitDefaults() that I can modify to limit the channels. I'm also seeing that FCC requires at least two channels.  Is there a cleaner way to config the channels?

**Reply by mluis1** [reply]
First I would like to inform that what you call a single channel gateway is not a gateway. In plus this kind of devices are not LoRaWAN compliant and may disrupt Network Servers operation.
I would strongly recommend to **not use** this kind of devices.

Although in order to answer the question on how to modify the default channels usage there is a better way.
Find below a code excerpt showing how to change the used channels on US915 and AU915 regions when 8 channels gateways are being used.
Restricting the channels usage on US915 and AU915 regions **should not be done** as these regions are supposed to use 64 channels gateways.

As a remark: Usually no file under `mac` directory requires to be modified.

Example:
```c
LmHandlerErrorStatus_t LmHandlerInit( LmHandlerCallbacks_t *handlerCallbacks,
                                      LmHandlerParams_t *handlerParams )
{
...

    // Try to restore from NVM and query the mac if possible.
    if( nbNvmData > 0 )
    {
        LmHandlerCallbacks->OnNvmDataChange( LORAMAC_HANDLER_NVM_RESTORE, nbNvmData );
    }
    else
    {
        // Configure the default datarate
        mibReq.Type = MIB_CHANNELS_DEFAULT_DATARATE;
        mibReq.Param.ChannelsDefaultDatarate = LmHandlerParams->TxDatarate;
        LoRaMacMibSetRequestConfirm( &mibReq );

        mibReq.Type = MIB_CHANNELS_DATARATE;
        mibReq.Param.ChannelsDatarate = LmHandlerParams->TxDatarate;
        LoRaMacMibSetRequestConfirm( &mibReq );

// BEGIN - For use with 8 channels gateway on US915 or AU915 regions
#if defined( REGION_US915 ) || defined( REGION_AU915 )
        // Enabling 1st block of 8 channels (0-7) + channel 64
        uint16_t channelMask[] = { 0x00FF, 0x0000, 0x0000, 0x0000, 0x0001, 0x0000};
        mibReq.Type = MIB_CHANNELS_MASK;
        mibReq.Param.ChannelsMask = channelMask;
        LoRaMacMibSetRequestConfirm( &mibReq );
        mibReq.Type = MIB_CHANNELS_DEFAULT_MASK;
        mibReq.Param.ChannelsDefaultMask = channelMask;
        LoRaMacMibSetRequestConfirm( &mibReq );
#endif
// END - For use with 8 channels gateway on US915 region

...
}
```

**Reply by shabaz-git** [reply]
Hi,

I am operating in the Indian frequency band (IN865). My requirement is after joining the network the node should work with a specific channel (like channel 4 starting from 0). How to achieve that? I tired by the below changes. 

changes made in LmHandlerInit() function
```
        uint16_t channelMask_new[] = { 0x0008};
        mibReq.Type = MIB_CHANNELS_MASK;
        mibReq.Param.ChannelsMask = channelMask_new;
        LoRaMacMibSetRequestConfirm( &mibReq );

        mibReq.Type = MIB_CHANNELS_DEFAULT_MASK;
        mibReq.Param.ChannelsDefaultMask = channelMask_new;
        LoRaMacMibSetRequestConfirm( &mibReq );
```
In LoRaMacStatus_t RegionIN865ChannelAdd( ChannelAddParams_t* channelAdd ) function commented the following line as this is changing the mask.

```
    RegionNvmGroup2->Channels[id].Band = 0;
    //RegionNvmGroup2->ChannelsMask[0] |= ( 1 << id );
    return LORAMAC_STATUS_OK;
```

But the node is unable to join the network and goes into a bad state as there is no default channel for processing the join request. Any corrections or additional changes need to be done?
